examples/cjs/cjs.py

```diff
@@ -57,24 +65,184 @@ def model(data):
                                 infer={"enumerate": "parallel"})
                 mu_y_t = rho * z
                 pyro.sample("y_{}".format(t), dist.Bernoulli(mu_y_t),
-                            obs=data[:, t])
-            first_capture_mask |= data[:, t].byte()
+                            obs=capture_history[:, t])
+            first_capture_mask |= capture_history[:, t].byte()
+
+
+"""
+In our second model variant there is a survival probability phi_t for 6
+of the 7 years of the capture data; each phi_t is treated as a fixed effect.
+"""
+
+
+def model_2(capture_history, sex):
+    N, T = capture_history.shape
+    rho = pyro.sample("rho", dist.Uniform(0.0, 1.0))  # recapture probability
+
+    z = torch.ones(N)
+    first_capture_mask = torch.zeros(N).byte()
+    # we create the plate once, outside of the loop over t
+    dippers_plate = pyro.plate("dippers", N, dim=-1)
+    for t in pyro.markov(range(T)):
+        # note that phi_t needs to be outside the plate, since
+        # phi_t is shared across all N birds
+        phi_t = pyro.sample("phi_{}".format(t), dist.Uniform(0.0, 1.0)) if t > 0 \
+                else 1.0
+        with dippers_plate, poutine.mask(mask=first_capture_mask):
+            mu_z_t = first_capture_mask.float() * phi_t * z + (1 - first_capture_mask.float())
+            # we use parallel enumeration to exactly sum out
+            # the discrete states z_t.
+            z = pyro.sample("z_{}".format(t), dist.Bernoulli(mu_z_t),
+                            infer={"enumerate": "parallel"})
+            mu_y_t = rho * z
+            pyro.sample("y_{}".format(t), dist.Bernoulli(mu_y_t),
+                        obs=capture_history[:, t])
+        first_capture_mask |= capture_history[:, t].byte()
+
+
+"""
+In our third model variant there is a survival probability phi_t for 6
+of the 7 years of the capture data (just like in model_2), but here
+each phi_t is treated as a random effect.
+"""
+
+
+def model_3(capture_history, sex):
+    def logit(p):
+        return torch.log(p / (1.0 - p))
```

Comment:
nit: for numerical stability use `return torch.log(p) - torch.log1p(-p)`

Reply:
done